addons/mail/models/discuss/ir_binary.py

```diff
@@ -21,4 +21,27 @@ def _find_record_check_access(self, record, access_token, field):
                     domain.append(("channel_member_ids", "any", [("guest_id", "=", current_guest.id)]))
                 if self.env["discuss.channel"].search_count(domain, limit=1):
                     return record.sudo()
+                if current_guest:
```

Comment:
I don't think this case should be specific to guest, (portal) users should also be affected.